client/state/invites/actions.js

```diff
@@ -83,3 +86,51 @@ export function resendInvite( siteId, inviteId ) {
 			} );
 	};
 }
+
+export function deleteInvite( siteId, inviteId ) {
+	return deleteInvites( siteId, [ inviteId ] );
+}
+
+export function deleteInvites( siteId, inviteIds ) {
```

Comment:
This API endpoint is a bit different in that it will almost always return "success", even if not all of the individual invite deletions succeed.

I am not sure whether that was the best design, but given that it's already in place, I think the approach taken here is solid:  dispatch a single success or failure action if the entire request succeeded or failed, otherwise dispatch both actions.